include/camp/resource.hpp

```diff
@@ -94,12 +94,36 @@ namespace resources
       void wait_for(Event *e) { m_value->wait_for(e); }
       void wait() { m_value->wait(); }
 
+      /*
+       * \brief Compares two Resources to see if they are equal.
+       *
+       * \return True or false depending on comparison with m_value if they are on the same platform.
+       */
+      bool operator==(Resource const& r)
+      {
+        if(get_platform() == r.get_platform()) {
+          return (*m_value == r);
+        }
+        return false;
+      }
+      /*
+       * \brief Compares two Resources to see if they are NOT equal.
+       *
+       * \return True or false depending on result of == operator.
+       */
+      bool operator!=(Resource const& r)
+      {
+        return !(*this == r);
+      }
+
     private:
       class ContextInterface
       {
       public:
         virtual ~ContextInterface() {}
         virtual Platform get_platform() const = 0;
+        virtual bool operator==(Resource const& r) = 0;
+        virtual bool operator!=(Resource const& r) = 0;
```

Comment:
It is best practice to use a virtual operator== or make a named method?

Reply:
Good question, I initially thought using `operator==` here would be best but since this is for a nested class that isn't interfacing with the users I think either would be fine. We don't actually need `operator!=` here so a named method might actually be cleaner in this one case.

Reply:
I agree with @mdavis36 on this

Reply:
Making these named methods makes other things freak out... so, still working on this one.